'Save Branch as Scene' also saves original scene

BenjaminRiley opened the pull request:
'Save Branch as Scene' would not mark the original scene as changed, and therefore no prompt would appear if the scene was closed immediately after saving the branch when no other changes had been made.

Fixes #32067

akien-mga commented:
I think this should be changed to use UndoRedo instead, which should increase the history version and mark the scene as modified, giving the option to undo the changes.

It probably doesn't make sense to undo saving the branch as a separate file (we won't go around and delete the created file), but the replacement of the in-scene branch by the new scene instance could be undone.

BenjaminRiley commented:
Fair enough. I'll give that a go.

BenjaminRiley commented:
I've mostly just wrapped the existing function calls in `add_do_`/`add_undo_`s and removed the `queue_delete()`. I hope that's how I'm supposed to do it.

akien-mga commented:
Looks good. I suppose you've confirmed that it works well (do, undo and redo)?

Can you squash commits together?

BenjaminRiley committed:
'Save Branch as Scene' adds to undo history

When saving a branch as a scene, can now undo replacing the original branch with the new branch scene. 'Save Branch as Scene' previously would not mark the original scene as changed, and therefore no prompt would appear if the scene was closed immediately after saving the branch.

The new branch scene is not deleted when undo-ing.

BenjaminRiley commented:
Do, undo, redo, and marking the scene as modified all appear to work, yeah.

akien-mga commented:
Thanks!
